packages/components/rating/stories/rating.stories.tsx

```diff
@@ -0,0 +1,289 @@
+    </div>
+  );
+};
+
+const PrecisionTemplate = (args) => {
+  const [value1, setValue1] = useState(0);
+  const [value2, setValue2] = useState(0);
+
+  return (
+    <div className="w-full flex flex-col gap-2">
+      <div className="flex gap-x-5">
+        <Rating precision={0.25} onValueChange={setValue1} {...args} length={5} />
+        <span className="text-xs text-foreground-500">Precision 0.25: {value1}</span>
+      </div>
+      <div className="flex gap-x-5">
+        <Rating precision={0.2} onValueChange={setValue2} {...args} length={5} />
+        <span className="text-xs text-foreground-500">Precision 0.20: {value2}</span>
+      </div>
+    </div>
+  );
+};
+
+const CustomIconTemplate = (args) => {
+  const [value1, setValue1] = useState("0");
+  const [value2, setValue2] = useState("0");
+  const [value3, setValue3] = useState("0");
+
+  return (
+    <div className="w-full flex flex-col gap-2">
+      <div className="flex gap-x-10">
+        <Rating
+          {...args}
+          fillColor="red"
+          icon={<HeartIcon />}
+          length={5}
+          onValueChange={setValue1}
+        />
+        <span className="text-tiny text-foreground-500">Heart Rating: {value1}</span>
+      </div>
+      <div className="flex gap-x-10">
+        <Rating
+          {...args}
+          fillColor="green"
+          icon={<LikeIcon />}
+          length={5}
+          onValueChange={setValue2}
+        />
+        <span className="text-tiny text-foreground-500">Like Rating: {value2}</span>
+      </div>
+      <div className="flex gap-x-10">
+        <Rating
+          {...args}
+          fillColor="blue"
+          icon={<MusicIcon />}
+          length={5}
+          onValueChange={setValue3}
+        />
+        <div className="text-tiny text-foreground-500">Music Rating: {value3}</div>
+      </div>
+    </div>
+  );
+};
+
+const CustomSegmentTemplate = (args) => {
+  const [value, setValue] = useState(0);
+
+  return (
+    <div>
+      <Rating isSingleSelection length={4} onValueChange={setValue}>
+        <RatingSegment fillColor="red" icon={<AngryEmojicon />} index={0} {...args} />
+        <RatingSegment fillColor="orange" icon={<SadEmojicon />} index={1} {...args} />
+        <RatingSegment icon={<StraightEmojicon />} index={2} {...args} />
+        <RatingSegment fillColor="green" icon={<HappyEmojicon />} index={3} {...args} />
+      </Rating>
+      <div className="text-tiny text-foreground-500">Rating Value: {value}</div>
+    </div>
+  );
+};
+
+export const Default = {
+  render: Template,
+  args: {
+    ...defaultProps,
+  },
+};
+
+export const Accessibility = {
+  render: Template,
+  args: {
+    ...defaultProps,
+    length: 5,
+    "aria-label": "Rate this item",
+  },
+  parameters: {
+    a11y: {disable: false},
+  },
+};
+
+export const Required = {
+  render: WithReactHookFormTemplate,
+  args: {
+    ...defaultProps,
+    isRequired: true,
+  },
+};
```

Comment:
Could you modify `required` and `aria-required` base on the `validationBehavior`?
I would like to be able to switch in storybook.

Reply:
Done in [374136d](https://github.com/nextui-org/nextui/pull/3808/commits/374136dec9ee9ecc498ad02463d1cd0cc0fa66c7)